pkg/stanza/fileconsumer/file.go

```diff
@@ -92,7 +90,11 @@ func (m *Manager) Stop() error {
 	m.wg.Wait()
 	m.closePreviousFiles()
 	if m.persister != nil {
-		if err := checkpoint.Save(context.Background(), m.persister, m.knownFiles); err != nil {
+		checkpoints := make([]*reader.Metadata, 0, m.totalReaders())
+		for _, knownFiles := range m.knownFiles {
+			checkpoints = append(checkpoints, knownFiles.Get()...)
+		}
```

Comment:
Another subtle benefit of inverting the order of knownFiles is that this list becomes ordered as well, so when searching through it after a restart, we look at newest first.